client/client.go

```diff
@@ -343,16 +346,132 @@ func configureAwsClient(ctx context.Context, logger hclog.Logger, awsConfig *Con
+	ListAccounts(ctx context.Context, params *organizations.ListAccountsInput, optFns ...func(*organizations.Options)) (*organizations.ListAccountsOutput, error)
+}
+
+func getOUAccounts(ctx context.Context, accountsApi ListAccountsAPI, ous []string) ([]orgTypes.Account, error) {
+	var rawAccounts []orgTypes.Account
+
+	for _, ou := range ous {
+		var paginationToken *string
+		for {
+			resp, err := accountsApi.ListAccountsForParent(ctx, &organizations.ListAccountsForParentInput{
+				NextToken: paginationToken,
+				ParentId:  aws.String(ou),
+			})
+			if err != nil {
+				return nil, err
+			}
+			rawAccounts = append(rawAccounts, resp.Accounts...)
+			if resp.NextToken == nil {
+				break
+			}
+			paginationToken = resp.NextToken
+		}
+	}
+
+	return rawAccounts, nil
+}
+
+func getAllAccounts(ctx context.Context, accountsApi ListAccountsAPI) ([]orgTypes.Account, error) {
+	var rawAccounts []orgTypes.Account
+	var paginationToken *string
+
+	for {
+		resp, err := accountsApi.ListAccounts(ctx, &organizations.ListAccountsInput{
+			NextToken: paginationToken,
+		})
+		if err != nil {
+			return nil, err
+		}
+		rawAccounts = append(rawAccounts, resp.Accounts...)
+		if resp.NextToken == nil {
+			break
+		}
+		paginationToken = resp.NextToken
+	}
+	return rawAccounts, nil
+
+}
+
+func loadAccounts(ctx context.Context, awsConfig *Config, accountsApi ListAccountsAPI) ([]Account, error) {
+
+	var rawAccounts []orgTypes.Account
+	var err error
+	if len(awsConfig.Organization.OrganizationUnits) > 0 {
+		rawAccounts, err = getOUAccounts(ctx, accountsApi, awsConfig.Organization.OrganizationUnits)
+	} else {
+		rawAccounts, err = getAllAccounts(ctx, accountsApi)
+	}
+
+	if err != nil {
+		return []Account{}, err
+	}
+	accounts := make([]Account, 0)
+	for _, account := range rawAccounts {
+		// Only load Active accounts
+		if account.Status != orgTypes.AccountStatusActive {
+			continue
+		}
+		localProfile := ""
+		if awsConfig.Organization.AdminAccount != nil {
+			localProfile = awsConfig.Organization.AdminAccount.LocalProfile
+		}
+		accounts = append(accounts, Account{
+			ID:              *account.Id,
+			RoleARN:         GenerateResourceARN("iam", "role", awsConfig.Organization.ChildAccountRoleName, "", *account.Id),
+			RoleSessionName: awsConfig.Organization.ChildAccountRoleSessionName,
+			ExternalID:      awsConfig.Organization.ChildAccountExternalID,
+			LocalProfile:    localProfile,
+			Regions:         awsConfig.Organization.ChildAccountRegions,
+			source:          "org",
+		})
+	}
+	return accounts, err
+}
+
+func loadOrgAccounts(ctx context.Context, logger hclog.Logger, awsConfig *Config) ([]Account, AssumeRoleAPIClient, error) {
+
+	adminAccount := Account{AccountName: "Default-Admin-Account"}
+	if awsConfig.Organization.AdminAccount != nil {
+		adminAccount = *awsConfig.Organization.AdminAccount
+	}
+	awsCfg, err := configureAwsClient(ctx, logger, awsConfig, adminAccount, nil)
+	if err != nil {
+		return nil, nil, err
+	}
```

Comment:
document this line, as it is hard to follow the logic here, took me a few minutes to fully understand ~ what is done here.